src/engine/execute.js

```diff
@@ -16,38 +16,42 @@ var isPromise = function (value) {
  * @param {!Thread} thread Thread which to read and execute.
  */
 var execute = function (sequencer, thread) {
+
     var runtime = sequencer.runtime;
     var target = thread.target;
 
-    // Current block to execute is the one on the top of the stack.
-    var currentBlockId = thread.peekStack();
-    var currentStackFrame = thread.peekStackFrame();
-
-    // Check where the block lives: target blocks or flyout blocks.
-    var targetHasBlock = (
-        typeof target.blocks.getBlock(currentBlockId) !== 'undefined'
-    );
-    var flyoutHasBlock = (
-        typeof runtime.flyoutBlocks.getBlock(currentBlockId) !== 'undefined'
-    );
-
     // Stop if block or target no longer exists.
-    if (!target || (!targetHasBlock && !flyoutHasBlock)) {
+    if (target === null) {
         // No block found: stop the thread; script no longer exists.
         sequencer.retireThread(thread);
         return;
     }
 
-    // Query info about the block.
-    var blockContainer = null;
-    if (targetHasBlock) {
-        blockContainer = target.blocks;
-    } else {
+    // Current block to execute is the one on the top of the stack.
+    var currentBlockId = thread.peekStack();
+    var currentStackFrame = thread.peekStackFrame();
+
+    var blockContainer = target.blocks;
+    var block = blockContainer.getBlock(currentBlockId);
+    if (typeof block === 'undefined') {
         blockContainer = runtime.flyoutBlocks;
+        block = blockContainer.getBlock(currentBlockId);
+        // Stop if block or target no longer exists.
+        if (typeof block === 'undefined') {
+            // No block found: stop the thread; script no longer exists.
+            sequencer.retireThread(thread);
+            return;
+        }
     }
-    var opcode = blockContainer.getOpcode(currentBlockId);
-    var fields = blockContainer.getFields(currentBlockId);
-    var inputs = blockContainer.getInputs(currentBlockId);
+
+    // if (eCount++ % 1000==0) {   // 603,000 times!!
+    //     console.log('Execute x' + eCount);
+    // }
+
+    //    var block = blockContainer.getBlock(currentBlockId);
+    var opcode = block.opcode; // blockContainer.getOpcode(currentBlockId);
+    var fields = block.fields; // blockContainer.getFields(currentBlockId);
```

Comment:
may be varying opinions here but if it were me I might consider removing the simple blocks getters (i.e., returning one property of one block) altogether. can't imagine these getters adding much, and it might make sense to specifically be constructing the vm code to be robust to direct access to block fields/properties like you did here (for performance as you've noticed :)) Having the getters and *also* directly accessing the properties during execution could be misleading...